### PR TITLE
Implement SupportsCompressor for migration cache

### PR DESCRIPTION
**Pull request by maggie-lou**
* Also fix a bug in FindMissing where we were hardcoding "Compressor: repb.Compressor_IDENTITY" when queuing the data to be copied. FindMissing is always called with resources with the identity compressor anyway, so this shouldn't matter. But it's technically incorrect and could cause a subtle bug later

### CODE REVIEW

**Comment by maggie-lou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2956** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2956" target="_blank">Graphite</a> 
    * **PR #2838** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2838" target="_blank">Graphite</a> 
      * **PR #2932** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2932" target="_blank">Graphite</a> 
      * **PR #2962** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2962" target="_blank">Graphite</a> 
        * **PR #2966** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2966" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2966?utm_source=stack-comment).

File: enterprise/server/backends/migration_cache/migration_cache.go
Comment (bduffany) [inline enterprise/server/backends/migration_cache/migration_cache.go:244]: This change LGTM but seems unrelated -- just leaving this comment as a sanity check in case you meant to include it in another PR instead.

File: enterprise/server/backends/migration_cache/migration_cache.go
Comment (bduffany) [inline enterprise/server/backends/migration_cache/migration_cache.go:842]: This LGTM at a glance, assuming the underlying cache impls returns a correct value of `SupportsCompressor` during a rollout. There might be other concerns I'm not thinking of though, so it'd be good to get an LGTM from @vadimberezniker and/or @tylerwilliams as well

**Review by bduffany — APPROVED** [review]

File: enterprise/server/backends/migration_cache/migration_cache.go
Comment (maggie-lou) [reply enterprise/server/backends/migration_cache/migration_cache.go:244]: Thanks for the call out. I'll add the following note to the commit description too to reduce confusion

I changed this because I realized I was hardcoding "Compressor: repb.Compressor_IDENTITY" when queuing the data to be copied below. FindMissing is always called with resources with the identity compressor anyway, so this shouldn't matter. But it's technically incorrect and could cause a subtle bug later

File: enterprise/server/backends/migration_cache/migration_cache.go
Comment (maggie-lou) [reply enterprise/server/backends/migration_cache/migration_cache.go:842]: Sounds good

[Distributed cache will return SupportsCompressor=false ](https://github.com/buildbuddy-io/buildbuddy/pull/2962/files#diff-9410f9cf8e952fbcc75b77758e162afd98b0225ec51988fe9e2254a911b34efaR1012)unless all the peers return SupportsCompressor=true. So during a rollout, distributed cache should return SupportsCompressor=false and the migration cache should return SupportsCompressor=false. Once the rollout has proceeded far enough for distributed SupportsCompressor=true, then the migration cache should also return SupportsCompressor=true

Comment (maggie-lou) [reply enterprise/server/backends/migration_cache/migration_cache.go]: @vadimberezniker I think this code should be safe for the roll-out with the distributed roll out plan we discussed in that other PR but could you help sanity check me and brandon?

**Review by vadimberezniker — APPROVED** [review]

**Pull request merged**
